SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/Operations/QueueManaging/BrokerProfileJobProvider.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  BrokerProfileJobProvider.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Common
+import Foundation
+
+public protocol BrokerProfileJobProviding {
+    func createJobs(with jobType: JobType,
+                    withPriorityDate priorityDate: Date?,
+                    showWebView: Bool,
+                    errorDelegate: BrokerProfileJobErrorDelegate,
+                    jobDependencies: BrokerProfileJobDependencyProviding) throws -> [BrokerProfileJob]
+}
+
+public final class BrokerProfileJobProvider: BrokerProfileJobProviding {
```

Comment:
This class was one that I proposed getting rid of, though I have to admit it does help a bit with testability. I renamed it for now to be consistent with other changes, but I think it could also be scrapped by moving this logic to somewhere else that is overridable, like the dependency provider. Not fussy either way, tbh I think this class is fairly benign anyway.